### PR TITLE
Prepare for v4.4.0

### PR DESCRIPTION
**Pull request by nchen63**
hold for 
- [x] #2106 
- [x] #1757 
- [x] #2102

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+* [new-rule] `strict-type-predicates` (#2046)
+* [new-rule] `misused-new` (#1963)
+* [new-rule] `prefer-method-signature` (#2028)
+* [new-rule] `prefer-function-over-method` (#2037)
+* [new-rule-option] `allow-fast-null-checks` added to `no-unused-expression` (#1638)
+* [new-rule-option] `comment-format-rule` adds `ignore-words` and `ignore-pattern` options (#1757)
+* [new-rule-option] `whitespace` adds `check-preblock` option (#2002)
+* [new-rule-option] `member-ordering` adds categories, allowing flexible ordering of different types of members (#2041)
+* [new-rule-option] `strict-boolean-expressions` adds `allow-null-union`, `allow-undefined-union`, `allow-string`, and `allow-number` and  (#2033)
+* [new-fixer] `align` (#2097)
```
Comment (ajafff) [inline CHANGELOG.md]: [new-fixer] `no-trailing-whitespace` (2060)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+* [bugfix] `no-magic-numbers` false positive on default parameter values (#2004)
+* [bugfix] `no-empty-interface` allow empty interface with 2 or more parents (#2070)
+* [bugfix] `no-trailing-whitespace` fixed for comments and EOF (#2060)
+* [bugfix] `no-empty` no longer fails for private or protected constructor
```
Comment (ajafff) [inline CHANGELOG.md]: PR 1976

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,36 @@
...
+* [bugfix] `tslint:disable`/`tslint-enable` now handles multiple rules and fixes what code is enabled/disabled (#2061) 
+* [bugfix] `no-inferrable-types` now validates property declarations (#2081)
+* [bugfix] `unified-signatures` false positive (#2016)
```
Comment (ajafff) [inline CHANGELOG.md]: [bugfix] `whitespace` finds all whitespace errors in JsxExpressions and TemplateExpressions (2036)
[bugfix] `comment-format` no more false positives in JsxText (2036)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,36 @@
...
+* [enhancement] `--test` option now accepts glob
```
Comment (ajafff) [inline CHANGELOG.md]: PR 2079

File: src/configs/latest.ts [hunk 5]
```diff
@@ -47,6 +47,20 @@ export const rules = {
     }],
     "typeof-compare": true,
     "unified-signatures": true,
+
+    // added in v4.4
+    "arrow-return-shorthand": true,
+    "no-unbound-method": true,
```
Comment (adidahiya) [inline src/configs/latest.ts]: We haven't enabled any other rules that require type information in this config and I don't think we should start now. We could introduce a separate config for type checking rules and perhaps reconsider for the next major release.

**Comment by nchen63** [comment]
should `prefer-function-over-method` be in latest? I think it would be a bit much for most people

**Comment by adidahiya** [comment]
@nchen63 agreed, I would leave it out of the built-in configs

File: src/configs/latest.ts [hunk 6]
```diff
@@ -47,6 +47,13 @@ export const rules = {
     }],
     "typeof-compare": true,
     "unified-signatures": true,
+
+    // added in v4.4
+    "arrow-return-shorthand": true,
+    "no-unnecessary-initializer": true,
+    "no-misused-new ": true,
+    "prefer-method-signature": true,
+    "prefer-function-over-method": false,
```
Comment (adidahiya) [inline src/configs/latest.ts]: don't need this line if you're setting it to disabled

File: src/configs/latest.ts [hunk 7]
```diff
@@ -47,6 +47,13 @@ export const rules = {
     }],
     "typeof-compare": true,
     "unified-signatures": true,
+
+    // added in v4.4
+    "arrow-return-shorthand": true,
+    "no-unnecessary-initializer": true,
+    "no-misused-new ": true,
+    "prefer-method-signature": true,
```
Comment (adidahiya) [inline src/configs/latest.ts]: let's remove this too; I think it's a little too opinionated and would fail a lot of our builds

**Pull request merged**